lib/IRGen/GenFunc.cpp

```diff
@@ -1987,6 +2237,8 @@ std::optional<StackAddress> irgen::emitFunctionPartialApplication(
   SmallVector<SILType, 4> argValTypes;
   SmallVector<ParameterConvention, 4> argConventions;
 
+  GenericContextScope scope(IGF.IGM, origType->getInvocationGenericSignature());
```

Comment:
Another instance of `GenericContextScope`.

If this indeed the right solution, can we guard it behind an autodiff type check (only do it if the type is autodiff)?

Reply:
@aschwaighofer The PR is not autodiff related per se, though it is required for autodiff as autodiff seems to be the only part of the compiler that produces such SIL currently. 

The PR is about codegen support of `partial_apply` for a co-routine (modify / read accessors for example). Check the testcases, they are all plain SIL, no autodiff is involved.

Coroutines are quite special inside compiler in many aspects. One particular thing is that yields should be substituted in the context of caller, and this causes lots of headaches everywhere.

Reply:
I tend to agree that GenericContextScope might not be the right solution here. This is why I wanted to have @rjmccall's opinion, how it would be the best way to codegen partial_apply for coroutines given all special cases here.